src/modules/markets/components/markets-list.jsx

```diff
@@ -1,34 +1,114 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import MarketPreview from 'modules/market/components/market-preview';
 import Paginator from 'modules/common/components/paginator';
 import NullStateMessage from 'modules/common/components/null-state-message';
 
 import getValue from 'utils/get-value';
+import isEqual from 'lodash/isEqual';
 
-const MarketsList = (p) => {
-  const nullMessage = 'No Markets Available';
-
-  return (
-    <article className="markets-list">
-      {p.markets.length ? p.markets.map((market) => {
-        const selectedShareDenomination = getValue(p, `scalarShareDenomination.markets.${market.id}`);
-        const shareDenominations = getValue(p, 'scalarShareDenomination.denominations');
-
-        return (
-          <MarketPreview
-            key={market.id}
-            loginAccount={p.loginAccount}
-            {...market}
-            selectedShareDenomination={selectedShareDenomination}
-            shareDenominations={shareDenominations}
+import debounce from 'utils/debounce';
+
+export default class MarketsList extends Component {
```

Comment:
General comment: `MarketsList` is notably more complex than before, and it has pagination logic scattered throughout.  This feels "off" to me, although I haven't studied the new pagination code enough to make any specific recommendations.